sklearn/utils/estimator_checks.py

```diff
@@ -117,8 +117,8 @@ def _yield_checks(name, estimator):
 def _yield_classifier_checks(name, classifier):
     tags = _safe_tags(classifier)
 
-    # test classifiers can handle non-array data
-    yield check_classifier_data_not_an_array
+    # test classifiers can handle non-array data and pandas objects
+    yield check_classifier_two_data_types
```

Comment:
two_data_types is not clear (especially as dtype = data type is something else entirely). not_an_array is still acceptable.